### PR TITLE
Fix cancel, add logging of serial output

### PR DESCRIPTION
**Pull request by adjackura**
Don't use context.WithCancel for handling stopping of workflows, this caused issues because the cleanup function relied on that context.
Stream serial port 1 to GCS when WaitForInstancesStopped is called
Remove steps that are better suited for subworkflows

### CODE REVIEW

File: daisy/workflow/step_wait_for_instances_stopped.go [hunk 1]
```diff
@@ -15,13 +15,58 @@
...
 	"fmt"
+	"path"
 	"sync"
+	"time"
 )
 
 // WaitForInstancesStopped is a Daisy WaitForInstancesStopped workflow step.
 type WaitForInstancesStopped []string
 
+func steamSerial(w *Workflow, name string) {
```
Comment (crunk1) [inline daisy/workflow/step_wait_for_instances_stopped.go]: s/steam/stream

File: daisy/workflow/step_wait_for_instances_stopped.go [hunk 2]
```diff
@@ -15,13 +15,58 @@
...
+	var buf bytes.Buffer
+	var seenErr bool
+	for {
+		tick := time.Tick(1 * time.Second)
+		select {
+		case <-w.Ctx.Done():
+		case <-tick:
+			resp, err := w.ComputeClient.GetSerialPortOutput(w.Project, w.Zone, name, 1, start)
+			if err != nil {
+				// This is most likely because the instance has stopped, don't log but try one more time.
```
Comment (crunk1) [inline daisy/workflow/step_wait_for_instances_stopped.go]: Would it be better to explicitly check to see if the instance stopped?

File: daisy/workflow/test_common.go [hunk 3]
```diff
@@ -71,15 +71,15 @@ func init() {
 }
 
 func testWorkflow() *Workflow {
-	ctx, cancel := context.WithCancel(context.Background())
+	cancel := make(chan struct{})
```
Comment (crunk1) [inline daisy/workflow/test_common.go]: Why not using the Cancel? I thought cancelling a "parent" context would cascade to "children" contexts, cancelling them as well. In short, how does cancel work?

File: daisy/workflow/workflow.go [hunk 4]
```diff
@@ -224,8 +217,8 @@ func (s *Step) wrapValidateError(e error) error {
 // Workflow is a single Daisy workflow workflow.
 type Workflow struct {
 	// Populated on New() construction.
-	Ctx    context.Context    `json:"-"`
-	Cancel context.CancelFunc `json:"-"`
+	Ctx    context.Context `json:"-"`
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: I can't remember, do these need to be public?

File: daisy/workflow/workflow.go [hunk 5]
```diff
@@ -315,26 +308,27 @@ func (w *Workflow) FromFile(file string) error {
...
 	}
 	w.logger.Print("Uploading sources")
 	if err := w.uploadSources(); err != nil {
 		w.logger.Printf("Error uploading sources: %v", err)
-		w.Cancel()
+		close(w.Cancel)
 		return err
 	}
 	w.logger.Print("Running workflow")
+	defer w.cleanup()
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: We could maybe even put this at the top of Run(). Say, for example, we want to clean up artifacts from uploadSources().

File: daisy/workflow/test_common.go (hunk 3 above)
Comment (adjackura) [inline daisy/workflow/test_common.go]: For canceling the running workflow it works fine, the issue is we also use this context for cleanup which happens after a cancel call.

File: daisy/workflow/workflow.go (hunk 4 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: Yeah, daisy.go uses them

File: daisy/workflow/workflow.go (hunk 5 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: Moved it right after validate in case we add any of that.

**Pull request merged**